src/composables/staking/useStaking.ts

```diff
@@ -0,0 +1,25 @@
+import { DecoratedPoolWithShares } from '@/services/balancer/subgraph/types';
+
+export enum StakeState {
+  CanStake = 'can_stake',
+  MaxStaked = 'max_staked',
+  NoGuage = 'no_guage'
+}
+
+export default function useStaking() {
+  // TODO INTEGRATE STAKING CONTRACT
+  function getStakeState(pool: DecoratedPoolWithShares) {
+    // just random logic for differentiating between stake states // TODO REPLACE
+    if (Number(pool.shares) >= 20000) {
+      return StakeState.MaxStaked;
+    } else if (Number(pool.shares) < 20000 && Number(pool.shares) >= 5000) {
+      return StakeState.CanStake;
+    } else {
+      return StakeState.NoGuage;
+    }
```

Comment:
Perhaps not relevant for now - but this function could be defined before `useStaking` and just returned there so it would be re-defined each time. (tho - if it depends on other composable, then its fine for it to be here)

Reply:
ye i agree